modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/gnirs/Gnirs.scala

```diff
@@ -0,0 +1,183 @@
+    expTime <- extractExposureTime(config)
+    coadds  <- extractCoadds(config)
+    readMode <- config.extract(INSTRUMENT_KEY / READ_MODE_PROP).as[ReadMode]
+    wellDepth <- config.extract(INSTRUMENT_KEY / WELL_DEPTH_PROP).as[WellDepth]
+  } yield DCConfig(expTime, coadds, readMode, wellDepth))
+    .leftMap(e => SeqexecFailure.Unexpected(ConfigUtilOps.explain(e)))
+
+  private def getCCConfig(config: Config): TrySeq[CCConfig] = config.extract(INSTRUMENT_KEY / OBSERVE_TYPE_PROP).as[String]
+    .leftMap(e => SeqexecFailure.Unexpected(ConfigUtilOps.explain(e))).flatMap{
+    case DARK_OBSERVE_TYPE => GnirsController.Dark.right
+    case BIAS_OBSERVE_TYPE => SeqexecFailure.Unexpected("Bias not supported for GNIRS").left
+    case _                 => getCCOtherConfig(config)
+  }
+
+  private def getCCOtherConfig(config: Config): TrySeq[CCConfig] = (for {
+    xdisp  <- config.extract(INSTRUMENT_KEY / CROSS_DISPERSED_PROP).as[CrossDispersed]
+    woll   <- config.extract(INSTRUMENT_KEY / WOLLASTON_PRISM_PROP).as[WollastonPrism]
+    mode   <- getCCMode(config, xdisp, woll)
+    slit   <- config.extract(INSTRUMENT_KEY / SLIT_WIDTH_PROP).as[SlitWidth]
+    slitOp = getSlit(slit)
+    camera <- config.extract(INSTRUMENT_KEY / CAMERA_PROP).as[Camera]
+    decker <- getDecker(config, slit, woll, xdisp)
+    filter <- config.extract(INSTRUMENT_KEY / FILTER_PROP).as[Filter].toOption.right
+    filter1 = getFilter1(filter, slit, decker)
+    filter2 <- getFilter2(config, filter, xdisp)
+  } yield Other(mode, camera, decker, filter1, filter2, slitOp) )
+    .leftMap(e => SeqexecFailure.Unexpected(ConfigUtilOps.explain(e)))
+
+  private def getCCMode(config: Config, xdispersed: CrossDispersed, woll: WollastonPrism): ConfigUtilOps.ExtractFailure\/GnirsController.Mode = for {
+    acq        <- config.extract(INSTRUMENT_KEY / ACQUISITION_MIRROR_PROP).as[AcquisitionMirror]
+    disperser  <- config.extract(INSTRUMENT_KEY / DISPERSER_PROP).as[Disperser]
+  } yield {
+    if(acq === AcquisitionMirror.IN) GnirsController.Acquisition
+    else xdispersed match {
+      case CrossDispersed.SXD => GnirsController.CrossDisperserS(disperser)
+      case CrossDispersed.LXD => GnirsController.CrossDisperserL(disperser)
+      case _                  => if(woll === WollastonPrism.YES) GnirsController.Wollaston(disperser)
+                                 else GnirsController.Mirror(disperser)
+    }
+  }
+
+  private def getDecker(config: Config, slit: SlitWidth, woll: WollastonPrism, xdisp: CrossDispersed): ConfigUtilOps.ExtractFailure\/GnirsController.Decker =
+    config.extract(INSTRUMENT_KEY / DECKER_PROP).as[Decker].orElse{
+      for {
+        pixScale <- config.extract((INSTRUMENT_KEY / PIXEL_SCALE_PROP)).as[PixelScale]
+      } yield xdisp match {
+        case CrossDispersed.LXD => Decker.LONG_CAM_X_DISP
+        case CrossDispersed.SXD => Decker.SHORT_CAM_X_DISP
+        case _                  => {
```

Comment:
This `{` is not needed

Reply:
I know, but prevents Idea from messing the indentation.